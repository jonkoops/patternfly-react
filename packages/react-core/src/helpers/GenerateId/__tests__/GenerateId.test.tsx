/**
 * @vitest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';
import { expect, test } from 'vitest';

import { GenerateId } from '../GenerateId';

test('generates id', () => {
  const { asFragment } = render(
    <GenerateId isRandom={false}>{(id) => <div id={id}>div with random ID</div>}</GenerateId>
  );

  expect(asFragment()).toMatchSnapshot();
});
