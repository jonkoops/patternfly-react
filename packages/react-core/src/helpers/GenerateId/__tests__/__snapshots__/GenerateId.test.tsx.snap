// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generates id 1`] = `
<DocumentFragment>
  <div
    id="pf-random-id-0"
  >
    div with random ID
  </div>
</DocumentFragment>
`;
